connect-inject/endpoints_controller.go

```diff
@@ -442,7 +448,86 @@ func (r *EndpointsController) Logger(name types.NamespacedName) logr.Logger {
 func (r *EndpointsController) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&corev1.Endpoints{}).
-		Complete(r)
+		Watches(
+			&source.Kind{Type: &corev1.Pod{}},
+			&handler.EnqueueRequestsFromMapFunc{ToRequests: handler.ToRequestsFunc(r.requestsForRunningAgentPods)},
+			builder.WithPredicates(predicate.NewPredicateFuncs(r.filterAgentPods)),
+		).Complete(r)
+}
+
+// filterAgentPods receives meta and object information for Kubernetes resources that are being watched,
+// which in this case are Pods. It only returns true if the Pod is a Consul Agent Pod. It reads the labels
+// from the meta of the resource and uses the values of the "app" and "component" label to validate that
+// the Pod is a Consul Agent.
+func (r EndpointsController) filterAgentPods(meta metav1.Object, object runtime.Object) bool {
+	labels := meta.GetLabels()
+	app, ok := labels["app"]
+	if !ok {
+		return false
+	}
+	component, ok := labels["component"]
+	if !ok {
+		return false
+	}
+	// todo: do we want to filter on other labels?
+	if app == "consul" && component == "client" {
+		return true
+	}
+	return false
+}
+
+// requestsForRunningAgentPods creates a slice of requests for the endpoints controller. It enqueues a request for each endpoint that needs to be reconciled.
+// It iterates through the list of endpoints and creates a request for those endpoints that have an address that are on the same node as the new Consul Agent pod.
+// It receives a Pod Object which is a Consul Agent that has been filtered by filterAgentPods and only processes endpoints for pods where the Ready condition is true.
```

Comment:
could we break this up so the lines aren't too long?

Reply:
done done done done